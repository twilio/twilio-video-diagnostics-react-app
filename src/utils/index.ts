import { AUDIO_LEVEL_THRESHOLD } from '../constants';

export function getAudioLevelPercentage(level: number) {
  return (level * 100) / AUDIO_LEVEL_THRESHOLD; // 0 to 100
}

export function getStandardDeviation(values: number[]): number {
  // Same method used in client sdks
  // https://github.com/twilio/twilio-client.js/blob/master/lib/twilio/statsMonitor.ts#L88

  if (values.length <= 0) {
    return 0;
  }

  const valueAverage: number =
    values.reduce((partialSum: number, value: number) => partialSum + value, 0) / values.length;

  const diffSquared: number[] = values.map((value: number) => Math.pow(value - valueAverage, 2));

  const stdDev: number = Math.sqrt(
    diffSquared.reduce((partialSum: number, value: number) => partialSum + value, 0) / diffSquared.length
  );

  return round(stdDev);
}

export const round = (num: number, decimals = 2) =>
  Math.round((num + Number.EPSILON) * 10 ** decimals) / 10 ** decimals;
