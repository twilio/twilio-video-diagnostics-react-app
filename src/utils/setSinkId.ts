/**
 * Helper function to detect whether or not the global `HTMLAudioElement`
 * constructor supports `setSinkId`.
 */
export function isSetSinkIdSupported() {
  return typeof window.Audio.prototype.setSinkId === 'function';
}
