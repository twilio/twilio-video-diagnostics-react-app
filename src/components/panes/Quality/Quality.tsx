import { ExcellentQuality } from './ExcellentQuality/ExcellentQuality';
import { PoorQuality } from './PoorQuality/PoorQuality';
import { getQualityScore } from './getQualityScore/getQualityScore';
import { useAppStateContext } from '../../AppStateProvider/AppStateProvider';
import { QualityModal } from './QualityModal/QualityModal';
import { useState } from 'react';

export enum QualityScore {
  Poor,
  Suboptimal,
  Good,
  Excellent,
}

export function Quality() {
  const { state } = useAppStateContext();
  const [isModalOpen, setIsModalOpen] = useState(false);
  const { totalQualityScore, latency, jitter, packetLoss, bitrate } = getQualityScore(
    state.preflightTest.report,
    state.bitrateTest.report
  );

  return (
    <>
      <QualityModal
        isModalOpen={isModalOpen}
        setIsModalOpen={setIsModalOpen}
        jitter={jitter}
        latency={latency}
        packetLoss={packetLoss}
        bitrate={bitrate}
      />
      {totalQualityScore === QualityScore.Good || totalQualityScore === QualityScore.Excellent ? (
        <ExcellentQuality quality={totalQualityScore} openModal={() => setIsModalOpen(true)} />
      ) : (
        <PoorQuality quality={totalQualityScore} openModal={() => setIsModalOpen(true)} />
      )}
    </>
  );
}
