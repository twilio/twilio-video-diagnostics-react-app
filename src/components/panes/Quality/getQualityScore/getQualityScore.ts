import { QualityScore } from '../Quality';
import { PreflightTestReport } from 'twilio-video';
import { MediaConnectionBitrateTest } from '@twilio/rtc-diagnostics';

export function getSingleQualityScore(
  stat: number | undefined,
  goodThreshold: number,
  suboptimalThreshold: number,
  poorThreshold: number,
  descending: boolean = false
) {
  if (typeof stat === 'undefined') {
    // We ignore values that are missing
    return QualityScore.Excellent;
  }

  if (descending) {
    if (stat > goodThreshold) return QualityScore.Excellent;
    if (stat > suboptimalThreshold) return QualityScore.Good;
    if (stat > poorThreshold) return QualityScore.Suboptimal;
    return QualityScore.Poor;
  }

  if (stat >= poorThreshold) return QualityScore.Poor;
  if (stat >= suboptimalThreshold) return QualityScore.Suboptimal;
  if (stat >= goodThreshold) return QualityScore.Good;
  return QualityScore.Excellent;
}

export const formatNumber = (val: number | undefined) => {
  return val?.toLocaleString(undefined, {
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  });
};

export function getQualityScore(
  preflightTestReport: PreflightTestReport | null,
  bitrateTestReport: MediaConnectionBitrateTest.Report | null
) {
  const maxBitrate = bitrateTestReport?.values ? Math.max(...bitrateTestReport.values) : 0;

  const latency = {
    average: formatNumber(preflightTestReport?.stats!.rtt!.average),
    max: formatNumber(preflightTestReport?.stats!.rtt!.max),
    qualityScore: getSingleQualityScore(preflightTestReport?.stats!.rtt!.average, 100, 250, 400),
  };

  const jitter = {
    average: formatNumber(preflightTestReport?.stats!.jitter!.average),
    max: formatNumber(preflightTestReport?.stats!.jitter!.max),
    qualityScore: getSingleQualityScore(preflightTestReport?.stats!.jitter!.average, 5, 10, 30),
  };

  const packetLoss = {
    average: formatNumber(preflightTestReport?.stats!.packetLoss!.average),
    max: formatNumber(preflightTestReport?.stats!.packetLoss!.max),
    qualityScore: getSingleQualityScore(preflightTestReport?.stats!.packetLoss!.average, 1, 3, 8),
  };

  const bitrate = {
    average: formatNumber(bitrateTestReport?.averageBitrate!),
    max: formatNumber(maxBitrate),
    qualityScore: getSingleQualityScore(bitrateTestReport?.averageBitrate!, 1000, 500, 150, true),
  };

  const totalQualityScore = Math.min(
    latency.qualityScore,
    jitter.qualityScore,
    packetLoss.qualityScore,
    bitrate.qualityScore
  ) as QualityScore;

  return {
    latency,
    jitter,
    packetLoss,
    bitrate,
    totalQualityScore,
  };
}
