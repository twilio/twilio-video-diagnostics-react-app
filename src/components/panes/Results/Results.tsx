import { makeStyles, Button, Container, Grid, Typography, Theme, createStyles, Hidden } from '@material-ui/core';
import { ActivePane, useAppStateContext } from '../../AppStateProvider/AppStateProvider';
import { CheckMark } from '../../../icons/CheckMark';
import { DownloadIcon } from '../../../icons/DownloadIcon';
import { getQualityScore } from '../Quality/getQualityScore/getQualityScore';
import { QualityScore } from '../Quality/Quality';
import { SmallError } from '../../../icons/SmallError';
import SomeFailed from './SomeFailed.png';
import TestsPassed from './TestsPassed.png';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    header: {
      float: 'left',
      [theme.breakpoints.down('md')]: {
        float: 'initial',
        paddingBottom: '1em',
      },
    },
    buttonContainer: {
      display: 'inline-flex',
      flexWrap: 'wrap',
      gap: '1em',
      width: '100%',
    },
    resultsList: {
      float: 'right',
      [theme.breakpoints.down('md')]: {
        float: 'initial',
      },
    },
    resultContainer: {
      marginTop: '1.5em',
      '&:not(:last-child)': {
        paddingBottom: '1.5em',
        borderBottom: '0.1em solid #8891AA',
      },
      [theme.breakpoints.down('md')]: {
        '&:last-child': {
          paddingBottom: '1.5em',
        },
      },
    },
    iconContainer: {
      display: 'flex',
      '& svg': {
        margin: '0.2em 0.8em 0 0',
      },
    },
    downloadButton: {
      '& svg': {
        position: 'relative',
        left: '-5px',
      },
    },
    restartButton: {
      backgroundColor: '#FFFFFF',
      borderColor: '#8891AA',
    },
    illustration: {
      marginTop: '7em',
    },
    hardwareButton: {
      marginRight: '1.5em',
    },
    gutterBottom: {
      marginBottom: '1em',
      [theme.breakpoints.down('md')]: {
        marginBottom: '1.5em',
      },
    },
  })
);

export function Results() {
  const { state, downloadFinalTestResults, dispatch } = useAppStateContext();
  const { totalQualityScore } = getQualityScore(state.preflightTest.report, state.bitrateTest.report);
  const classes = useStyles();

  const testsPassed = totalQualityScore === QualityScore.Excellent || totalQualityScore === QualityScore.Good;
  const qualityScore = QualityScore[totalQualityScore].toLowerCase();

  return (
    <>
      <Container>
        <div>
          <Grid item lg={5} className={classes.header}>
            <Typography variant="h1" gutterBottom>
              {testsPassed ? 'All tests passed!' : 'Some tests failed'}
            </Typography>

            {testsPassed ? (
              <Typography variant="body1" gutterBottom>
                As far as we can tell, your video should be working. If you're still experiencing issues, download your
                report results and send to your network administrator.
              </Typography>
            ) : (
              <Typography variant="body1" gutterBottom>
                <strong>One out of three </strong>
                tests failed – use this list to solve common video issues and restart the test. If you can’t easily
                solve the problem(s), download report results and send to your network administrator.
              </Typography>
            )}

            <div className={classes.buttonContainer}>
              <Button
                variant="contained"
                color="primary"
                className={classes.downloadButton}
                onClick={downloadFinalTestResults}
              >
                <DownloadIcon />
                Download report results
              </Button>
              <Button variant="outlined" className={classes.restartButton} onClick={() => window.location.reload()}>
                Restart test
              </Button>
            </div>

            <Hidden mdDown>
              <img
                src={testsPassed ? TestsPassed : SomeFailed}
                alt={testsPassed ? 'Success' : 'Some Failed'}
                className={classes.illustration}
              />
            </Hidden>
          </Grid>

          <Grid item lg={5} className={classes.resultsList}>
            <div className={classes.resultContainer}>
              <div className={classes.iconContainer}>
                <CheckMark />
                <Typography variant="h3" className={classes.gutterBottom}>
                  Device &amp; Network Setup
                </Typography>
              </div>
              <Typography variant="body1" className={classes.gutterBottom}>
                Audio and video successfully received from your hardware and browser.
              </Typography>

              <div className={classes.buttonContainer}>
                <Button
                  variant="outlined"
                  onClick={() => dispatch({ type: 'set-active-pane', newActivePane: ActivePane.CameraTest })}
                >
                  Review hardware
                </Button>
                <Button
                  variant="outlined"
                  onClick={() => dispatch({ type: 'set-active-pane', newActivePane: ActivePane.BrowserTest })}
                >
                  Review browser
                </Button>
              </div>
            </div>

            <div className={classes.resultContainer}>
              <div className={classes.iconContainer}>
                <CheckMark />
                <Typography variant="h3" className={classes.gutterBottom}>
                  Connectivity
                </Typography>
              </div>
              <Typography variant="body1" className={classes.gutterBottom}>
                All connections are working successfully.
              </Typography>
              <Button
                variant="outlined"
                onClick={() => dispatch({ type: 'set-active-pane', newActivePane: ActivePane.Connectivity })}
              >
                Review connectivity
              </Button>
            </div>

            <div className={classes.resultContainer}>
              <div className={classes.iconContainer}>
                {testsPassed ? <CheckMark /> : <SmallError />}
                <Typography variant="h3" className={classes.gutterBottom}>
                  Quality &amp; Performance
                </Typography>
              </div>

              {testsPassed ? (
                <Typography variant="body1" className={classes.gutterBottom}>
                  Awesome! Your expected call quality is <strong>{qualityScore}</strong> and overall performance looks
                  {qualityScore === 'excellent' ? ' good' : ' ok'}.
                </Typography>
              ) : (
                <Typography variant="body1" className={classes.gutterBottom}>
                  Your overall score is <strong>{qualityScore}</strong> which means that your connection isn't good
                  enough to run video properly. Try out these tips and rerun the test.
                </Typography>
              )}

              <Button
                variant="outlined"
                onClick={() => dispatch({ type: 'set-active-pane', newActivePane: ActivePane.Quality })}
              >
                Review performance
              </Button>
            </div>
          </Grid>
        </div>
      </Container>
    </>
  );
}
