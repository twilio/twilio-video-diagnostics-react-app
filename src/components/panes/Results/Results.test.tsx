import { Button } from '@material-ui/core';
import { shallow } from 'enzyme';
import * as utils from '../../../utils';
import { ActivePane, useAppStateContext } from '../../AppStateProvider/AppStateProvider';
import { Results } from './Results';
import { getQualityScore } from '../Quality/getQualityScore/getQualityScore';
import { QualityScore } from '../Quality/Quality';

// @ts-ignore
utils.downloadJSONFile = jest.fn();

//@ts-ignore
delete window.location;

//@ts-ignore
window.location = { reload: jest.fn() };

jest.mock('../../AppStateProvider/AppStateProvider');
jest.mock('../Quality/getQualityScore/getQualityScore');

const mockUseAppStateContext = useAppStateContext as jest.Mock<any>;
const mockGetQualityScore = getQualityScore as jest.Mock<any>;

const mockDispatch = jest.fn();
const mockDownloadFinalTestResults = jest.fn();

mockUseAppStateContext.mockImplementation(() => ({
  state: { preflightTest: { report: 'mockReport' }, bitrateTest: { report: 'mockReport' } },
  dispatch: mockDispatch,
  downloadFinalTestResults: mockDownloadFinalTestResults,
}));

describe('the GetResults component', () => {
  it('should render correctly when score is "excellent"', () => {
    mockGetQualityScore.mockImplementationOnce(() => ({
      totalQualityScore: QualityScore.Excellent,
    }));
    const wrapper = shallow(<Results />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when score is "good"', () => {
    mockGetQualityScore.mockImplementationOnce(() => ({
      totalQualityScore: QualityScore.Good,
    }));
    const wrapper = shallow(<Results />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly when score is "suboptimal"', () => {
    mockGetQualityScore.mockImplementationOnce(() => ({
      totalQualityScore: QualityScore.Suboptimal,
    }));
    const wrapper = shallow(<Results />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when score is "poor"', () => {
    mockGetQualityScore.mockImplementationOnce(() => ({
      totalQualityScore: QualityScore.Poor,
    }));
    const wrapper = shallow(<Results />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('the button clicks', () => {
    beforeEach(() => {
      mockGetQualityScore.mockImplementationOnce(() => ({
        totalQualityScore: QualityScore.Good,
      }));
    });
    it('should download the final results report when "Download report results" is clicked on', () => {
      const wrapper = shallow(<Results />);
      wrapper.find(Button).at(0).simulate('click');

      expect(mockDownloadFinalTestResults).toHaveBeenCalled();
    });

    it('should refresh the page when "Restart test" is clicked on', () => {
      const wrapper = shallow(<Results />);
      wrapper.find(Button).find({ children: 'Restart test' }).simulate('click');

      expect(window.location.reload).toHaveBeenCalled();
    });

    it('should take user to the camera test when "Review hardware" is clicked on', () => {
      const wrapper = shallow(<Results />);
      wrapper.find(Button).find({ children: 'Review hardware' }).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ type: 'set-active-pane', newActivePane: ActivePane.CameraTest });
    });

    it('should take user to the browser check when "Review browser" is clicked on', () => {
      const wrapper = shallow(<Results />);
      wrapper.find(Button).find({ children: 'Review browser' }).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ type: 'set-active-pane', newActivePane: ActivePane.BrowserTest });
    });

    it('should take user to the connectivity check when "Review connectivity" is clicked on', () => {
      const wrapper = shallow(<Results />);
      wrapper.find(Button).find({ children: 'Review connectivity' }).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ type: 'set-active-pane', newActivePane: ActivePane.Connectivity });
    });

    it('should take user to the quality check when "Review performance" is clicked on', () => {
      const wrapper = shallow(<Results />);
      wrapper.find(Button).find({ children: 'Review performance' }).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ type: 'set-active-pane', newActivePane: ActivePane.Quality });
    });
  });
});
