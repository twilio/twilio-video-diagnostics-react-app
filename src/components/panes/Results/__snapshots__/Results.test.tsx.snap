// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GetResults component should render correctly when score is "excellent" 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="h1"
        >
          All tests passed!
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="body1"
        >
          As far as we can tell, your video should be working. If you're still experiencing issues, download your report results and send to your network administrator.
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-downloadButton-3"
          color="primary"
          onClick={[MockFunction]}
          variant="contained"
        >
          <DownloadIcon />
          Download report results
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8891AA",
            }
          }
          variant="outlined"
        >
          Restart test
        </WithStyles(ForwardRef(Button))>
        <img
          alt="Success"
          src="test-file-stub"
          style={
            Object {
              "marginTop": "7em",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Device & Network Setup
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Audio and video successfully received from your hardware and browser.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            style={
              Object {
                "marginRight": "1.5em",
              }
            }
            variant="outlined"
          >
            Review hardware
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review browser
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Connectivity
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            All connections are working successfully.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review connectivity
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Quality & Performance
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Awesome! Your expected call quality is 
            <strong>
              excellent
            </strong>
             and overall performance looks
             good
            .
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review performance
          </WithStyles(ForwardRef(Button))>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;

exports[`the GetResults component should render correctly when score is "good" 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="h1"
        >
          All tests passed!
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="body1"
        >
          As far as we can tell, your video should be working. If you're still experiencing issues, download your report results and send to your network administrator.
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-downloadButton-3"
          color="primary"
          onClick={[MockFunction]}
          variant="contained"
        >
          <DownloadIcon />
          Download report results
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8891AA",
            }
          }
          variant="outlined"
        >
          Restart test
        </WithStyles(ForwardRef(Button))>
        <img
          alt="Success"
          src="test-file-stub"
          style={
            Object {
              "marginTop": "7em",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Device & Network Setup
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Audio and video successfully received from your hardware and browser.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            style={
              Object {
                "marginRight": "1.5em",
              }
            }
            variant="outlined"
          >
            Review hardware
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review browser
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Connectivity
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            All connections are working successfully.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review connectivity
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Quality & Performance
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Awesome! Your expected call quality is 
            <strong>
              good
            </strong>
             and overall performance looks
             ok
            .
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review performance
          </WithStyles(ForwardRef(Button))>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;

exports[`the GetResults component should render correctly when score is "poor" 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="h1"
        >
          Some tests failed
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="body1"
        >
          <strong>
            One out of three 
          </strong>
          tests failed – use this list to solve common video issues and restart the test. If you can’t easily solve the problem(s), download report results and send to your network administrator.
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-downloadButton-3"
          color="primary"
          onClick={[MockFunction]}
          variant="contained"
        >
          <DownloadIcon />
          Download report results
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8891AA",
            }
          }
          variant="outlined"
        >
          Restart test
        </WithStyles(ForwardRef(Button))>
        <img
          alt="Some Failed"
          src="test-file-stub"
          style={
            Object {
              "marginTop": "7em",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Device & Network Setup
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Audio and video successfully received from your hardware and browser.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            style={
              Object {
                "marginRight": "1.5em",
              }
            }
            variant="outlined"
          >
            Review hardware
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review browser
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Connectivity
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            All connections are working successfully.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review connectivity
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <SmallError />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Quality & Performance
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Your overall score is 
            <strong>
              poor
            </strong>
             which means that your connection isn't good enough to run video properly. Try out these tips and rerun the test
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review performance
          </WithStyles(ForwardRef(Button))>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;

exports[`the GetResults component should render correctly when score is "suboptimal" 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justifyContent="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="h1"
        >
          Some tests failed
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          variant="body1"
        >
          <strong>
            One out of three 
          </strong>
          tests failed – use this list to solve common video issues and restart the test. If you can’t easily solve the problem(s), download report results and send to your network administrator.
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-downloadButton-3"
          color="primary"
          onClick={[MockFunction]}
          variant="contained"
        >
          <DownloadIcon />
          Download report results
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#8891AA",
            }
          }
          variant="outlined"
        >
          Restart test
        </WithStyles(ForwardRef(Button))>
        <img
          alt="Some Failed"
          src="test-file-stub"
          style={
            Object {
              "marginTop": "7em",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={5}
      >
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Device & Network Setup
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Audio and video successfully received from your hardware and browser.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            style={
              Object {
                "marginRight": "1.5em",
              }
            }
            variant="outlined"
          >
            Review hardware
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review browser
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <CheckMark />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Connectivity
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            All connections are working successfully.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review connectivity
          </WithStyles(ForwardRef(Button))>
        </div>
        <div
          className="makeStyles-resultContainer-1"
        >
          <div
            className="makeStyles-iconContainer-2"
          >
            <SmallError />
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h3"
            >
              Quality & Performance
            </WithStyles(ForwardRef(Typography))>
          </div>
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="body1"
          >
            Your overall score is 
            <strong>
              suboptimal
            </strong>
             which means that your connection isn't good enough to run video properly. Try out these tips and rerun the test
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            variant="outlined"
          >
            Review performance
          </WithStyles(ForwardRef(Button))>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
